src/reflect/type.go

```diff
@@ -433,10 +438,11 @@ type arrayType struct {
 
 type mapType struct {
 	rawType
-	numMethod uint16
-	ptrTo     *rawType
-	elem      *rawType
-	key       *rawType
+	extraFlags uint8
```

Comment:
All the other types that have an `elem` pointer match up with the `elemType` base struct (I suppose I could have embedded it, but I didn't ...).  Can you add the `extraFlags` field there as well as to the other elemType structs?

Reply:
Oh, I think you are right. This would be a problem if the struct doesn't have any padding (like on AVR - not a big platform but still kind of a lurking bug). I'll look into a fix.